src/components/cursor-controller.js

```diff
@@ -2,451 +2,190 @@ const TARGET_TYPE_NONE = 1;
-    this.data.playerRig.removeEventListener(this.data.releaseEvent, this._handlePrimaryUp);
-    this.data.playerRig.removeEventListener("cardboardbuttondown", this._handlePrimaryDown);
-    this.data.playerRig.removeEventListener("cardboardbuttonup", this._handlePrimaryUp);
+    return function() {
+      if (!this.enabled) {
+        return;
+      }
 
-    this.data.playerRig.removeEventListener("model-loaded", this._handleModelLoaded);
+      if (this.data.useMousePos) {
+        this.setRaycasterWithMousePos();
+      } else {
+        rayObjectRotation.setFromRotationMatrix(this.rayObject.matrixWorld);
+        this.direction
+          .set(0, 0, 1)
+          .applyQuaternion(rayObjectRotation)
+          .normalize();
+        this.origin.setFromMatrixPosition(this.rayObject.matrixWorld);
+        this.el.setAttribute("raycaster", { origin: this.origin, direction: this.direction });
+      }
 
-    this.el.sceneEl.removeEventListener("controllerconnected", this._handleControllerConnected);
-    this.el.sceneEl.removeEventListener("controllerdisconnected", this._handleControllerDisconnected);
-  },
+      const isGrabbing = this.data.cursor.components["super-hands"].state.has("grab-start");
+      if (isGrabbing) {
+        const distance = Math.min(
+          this.data.maxDistance,
+          Math.max(this.data.minDistance, this.currentDistance - this.currentDistanceMod)
+        );
+        this.direction.multiplyScalar(distance);
+        this.data.cursor.object3D.position.addVectors(this.origin, this.direction);
+      } else {
+        this.currentDistanceMod = 0;
+        this.updateDistanceAndTargetType();
+
+        const isTarget = this._isTargetOfType(TARGET_TYPE_INTERACTABLE_OR_UI);
+        if (isTarget && !this.wasCursorHovered) {
+          this.wasCursorHovered = true;
+          this.data.cursor.setAttribute("material", { color: this.data.cursorColorHovered });
+        } else if (!isTarget && this.wasCursorHovered) {
+          this.wasCursorHovered = false;
+          this.data.cursor.setAttribute("material", { color: this.data.cursorColorUnhovered });
+        }
+      }
 
-  tick: function() {
-    //handle physical hand
-    if (this.physicalHand) {
-      const state = this.physicalHand.components["super-hands"].state;
-      const isPhysicalHandGrabbing = state.has("grab-start") || state.has("hover-start");
-      if (this.wasPhysicalHandGrabbing != isPhysicalHandGrabbing) {
-        this._setCursorVisibility(!isPhysicalHandGrabbing);
-        this.currentTargetType = TARGET_TYPE_NONE;
+      if (this.data.drawLine) {
+        this.el.setAttribute("line", { start: this.origin.clone(), end: this.data.cursor.object3D.position.clone() });
       }
-      this.wasPhysicalHandGrabbing = isPhysicalHandGrabbing;
-      if (isPhysicalHandGrabbing) return;
-    }
+    };
+  })(),
 
-    //set raycaster origin/direction
+  setRaycasterWithMousePos() {
```

Comment:
This (and the following methods) should be setRaycasterWithMousePos: function() {`, no?

Reply:
👍